feat: Add to calendar other calendars and icons #25

roman-bytes opened the pull request:
Includes an Add to calendar button for commonly used calendars. 

I have not tested with Yahoo calendar since I do not have a yahoo email address or calendar.

BenDMyers commented on src/_includes/layouts/event.11ty.js:
```diff
@@ -0,0 +1,54 @@
+						</a>
+					</li>
+				</ul>
+				${content}
```
I noticed some of our styles weren't being applied, for things like margins between paragraphs, and I think it's because those styles are expecting all the content to be direct children of the `main`. Would you mind moving `content` right outside of this `div`?

```
    </ul>
</div>
${content}
```

roman-bytes replied:
Good catch, that did it.

chantastic commented:
@roman-bytes this is amazing!

i'm delighted by how quickly and completely you got this functionality added 🥳

i'm still new to Eleventy. so, i'm just going to say what i interpret from the code.

it looks like there is now a layer that "wraps" the event.html template to provide the calendar events.

i'm wondering if it's possible to handle this as an additional include — something like `event-calendar-links.html`. my inclination is that this might be a more intuitive path for code discovery in debugging.

thoughts?

roman-bytes commented:
@chantastic This makes way more sense, and shows my lack of Eleventy knowledge. This was actually my first attempt at using it at all. I have switched things around and things still seem to work. Will check a few more things and updated.

After this little exercise, I am starting to see why people like llty! 🎉

chantastic commented:
awesome! let's pick up discussion of @BenDMyers's concern here. Context: https://github.com/LunchDevCommunity/community-calendar/pull/38#discussion_r560589415

i agree with Ben — philosophically — that data and presentation should be separate.
that said, I also like to merge helpful features as soon as they are functionally ready, and follow-up as appetite demands.

any concerns with merging this munged (for the benefit of those wanting to add events to their calendars) while we work out a more correct solution in a follow-up branch?

chantastic commented:
as a counter-point to the philosophy of keeping data and template separate, i find value in keeping this interdependent code bundled together (for now)…

my justification is that — right now — the feature completely contained to one.
this might not be an ideal separation of concerns *but* it's clear which file to change for an update — there's only one!

at this point, we can only theorize what other uses of `googleCalendarEvent` or widespread SVG usage would look like.
if we separate those out now — without additional data points — we risk adding complexity for only a philosophical win.

this is just a counter point to provide guidance for future changes.
i'm not sure i love the alternative i'm imagining now. which is to divine a protocol for communicating to views which consumer, format, and icon should be used — an implementation that would be spread across multiple files.

roman-bytes commented:
I have no problem getting this merged in and then following it up with a better solution. 

Both of these are great points and I am not totally sure which is the "better" solution for us. I tend to lean towards @BenDMyers philosophy of keeping data and templates separate. While we can only theorize about other uses of things being broken out. Further down the road when that does happen, it will be ready for us.

So I say let's merge and maybe turn the follow-up branch into a lunch.dev session and explain why we went with that approach. Thoughts?

BenDMyers commented:
I like the idea of a lunch.dev session weighing pros and cons. @roman-bytes - Would you be willing to lead that session, and to pick a date?

roman-bytes commented:
@BenDMyers Never lead anything like that but I can give it a shot.

BenDMyers commented:
@all-contributors Please add @roman-bytes for code and event organizing!
